code/game/objects/items/robot/ai_upgrades.dm

```diff
@@ -18,9 +19,18 @@
 	else
 		to_chat(AI, "<span class='userdanger'>[user] has upgraded you with combat software!</span>")
 		AI.add_malf_picker()
+		if(announce_installation)
+			GLOB.minor_announcement.Announce("ERROR ER0RR $R-  [AI] has been upgraded with combat software. Thank you for your- 0RRO$!R41.%%!!(%$^^__+ @#F0E4.")
 	to_chat(user, "<span class='notice'>You upgrade [AI]. [src] is consumed in the process.</span>")
 	qdel(src)
 
+/obj/item/malf_upgrade/examine(mob/user)
+	. = ..()
+	if(isAntag(user))
+		if(announce_installation)
+			. += "<span class='warning'>You're familiar with this kind of illegal tech. You think you can disable the installation announcement by reprogramming \the [src] on an AI system integrity restorer console.</span>"
+		else
+			. += "<span class='warning'>The installation announcement on \the [src] was disabled.</span>"
```

Comment:
Slight grammar thing here. If you're examining this, surely `is` would make more sense than `was`. Also `notice` may be a better class here, instead of `warning`